resolver/resolver_test.go

```diff
@@ -338,3 +360,38 @@ func TestMultiError(t *testing.T) {
 		t.Fatalf("expected %q instead of %q", expected, actual)
 	}
 }
+
+func TestTruncate(t *testing.T) {
+	tm := newTruncated()
+	if !tm.Truncated {
+		t.Fatal("Message not truncated")
+	}
+	if l := tm.Len(); l > 512 {
+		t.Fatalf("Message to large: %d bytes", l)
+	}
+}
```

Comment:
It might be nice to test that adding one additional answer record via `genA(1)` brings the message length over 512.